docs/proposals/status.md

```diff
@@ -0,0 +1,126 @@
+## Background
+Application CRD has a list of GroupKinds and a LabelSelector to logically group live objects in the cluster. The Application CRD also provides a way to attach descriptive metadata to the logical grouping. 
+
+## What does this solve
+This PR proposes to enhance Application CRD controller to provide status tracking for the objects that match the selectors. This dramatically simplifies the status tracking requirements for a client that uses Application CRD. 
+
+## Proposal
+The proposal is to add a reconciler that creates watches for the GroupKinds that the application CRD refers to. The reconciler then inspects the matching objects and updates the aggregate status in the Application CRD.
+
+### New fields in Application CRD
+We propose to use Conditions to aggregate status of the matching objects.
+These conditions could be added to `.status.conditions`
+```
+"Ready"   //Aggregate readiness
+"Settled" //Observed generation is latest, the controller has acted on latest version of spec
+```
+In addition we could have the matching objects status also recorded in `.status.components`.  The motivation behind this is to provide individual component status as well for clients that need more than just the aggregate status. Well known object types can be aggregated to a higher degree of fidelity. For other types and custom resources we can propose standardizing `Ready` condition that could then be aggregated.
+
+```
+package application
+// ApplicationStatus defines controllers the observed state of Application
+type ApplicationStatus struct {
+	// ObservedGeneration is used by the Application Controller to report the last Generation of an Application
+	// that it has observed.
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	// Conditions - Ensure Ready,Settled conditions are set
+	Conditions []status.Condition   `json:"conditions,omitempty"`
+	// List of subcomponent status
+	Resources status.ResourceStatus `json:",inline"`
```

Comment:
Is there a benefit to a String rather than a bool.

Reply:
Resources is not a string.